pwa/src/templates/componentResult/ComponentResultsTemplate.tsx

```diff
@@ -0,0 +1,107 @@
+  const { t } = useTranslation();
+
+  switch (type) {
+    case "table":
+      return (
+        <Table>
+          <TableBody>
+            <TableRow>
+              <TableHeader>{t("Name")}</TableHeader>
+              <TableHeader>{t("Status")}</TableHeader>
+              <TableHeader>{t("Type")}</TableHeader>
+              <TableHeader>{t("Layer")}</TableHeader>
+              <TableHeader>{t("Organisation")}</TableHeader>
+              <TableHeader />
+            </TableRow>
+            {results.map((component) => (
+              <TableRow
+                key={component.id}
+                className={styles.contentRow}
+                onClick={() => navigate(`/components/${component.id}`)}
+              >
+                <TableCell>{component.name}</TableCell>
+
+                <TableCell>
+                  <Tag tag={component.developmentStatus} />
+                </TableCell>
+                <TableCell>
+                  <Tag tag={component.softwareType} />
+                </TableCell>
+
+                <TableCell>
+                  <Tag tag={_.upperFirst(component.intendedAudience)} />
+                </TableCell>
+
+                <TableCell>
+                  <Tag tag="Conduction" />
+                </TableCell>
+
+                <TableCell onClick={() => navigate(`/components/${component.id}`)}>
+                  <Link icon={<ArrowRightIcon />} iconAlign="start">
+                    {t("Details")}
+                  </Link>
+                </TableCell>
+              </TableRow>
+            ))}
+          </TableBody>
+        </Table>
+      );
+
+    case "cards":
+      return (
+        <div className={styles.componentsGrid}>
+          {results.map((component) => (
+            <RichContentCard
+              key={component.id}
+              link={{ label: component.name, href: `/components/${component.id}` }}
+              labelsWithIcon={[
+                { label: _.upperFirst(component.intendedAudience), icon: <HamburgerIcon /> },
+                { label: "Conduction", icon: <HouseIcon /> },
+              ]}
+              tags={[component.developmentStatus, component.softwareType]}
+              contentLinks={[
+                {
+                  title: t("Repository"),
+                  subTitle: t("Check out the repository on GitHub"),
+                  href: component.isBasedOn,
+                },
+              ]}
+            />
+          ))}
+        </div>
+      );
+    case "layer":
+      return (
+        <div className={styles.componentsGrid}>
+          {results.map((component) => (
+            <Card
+              key={component.id}
+              title={component.name}
+              date={new Date(component.releaseDate)}
+              subTitle={component.intendedAudience}
+              onClick={() => navigate(`/components/${component.id}`)}
+              variant="case"
+            />
+          ))}
+        </div>
+      );
```

Comment:
This works great. However, I think with a small adjustment, we can manage it much more easily.

I would create three new components **within** this component file. So completely outside of your first component declaration (so after the final `};` of this file), I'd make three new components, like so:

```
const Table: React.FC = () => {
  ... all table code here
}

const Cards: React.FC = () => {
  ... all table code here
}

const Layers: React.FC = () => {
  ... all table code here
}
```

Thereafter, you can just initiate these components in your switch, like so:

```
...
case "table":
      return <Table {...{components}} />
...
```